crates/sui-faucet/src/faucet/simple_faucet.rs

```diff
@@ -95,7 +93,14 @@ impl SimpleFaucet {
             {
                 let uuid = Uuid::from_bytes(uuid);
                 info!(?uuid, ?recipient, ?coin_id, "Retrying txn from WAL.");
-                pending.push((uuid, recipient, coin_id, tx));
+
+                let current_length = pending.len();
+                // group all coins that had the same uuid in a vector
+                if current_length != 0 && pending[current_length - 1].0 == uuid {
```

Comment:
Do we have some kind of guarantee that the pending requests in the WAL for the same UUID are next to each other?  If not, we should be using a map for this instead of a vector.